### ISSUE TITLE
Lab 08 Test Case 9_09

### ISSUE THREAD

**Issue by yqlok17** [nus-cs2030/2223-s1#340]
# Summary

Not able to pass this test case as it seems some of my customers are waiting forever.

## Description:

Waiting customers are not being served for some reason. From the statistics (screenshots below) it seems that some customers are not being dequeued properly and are waiting forever in the queue as a result (771 + 229 = 1000 compared to my 762 + 229 = 991).

For context, my logic works as such for waiting:

Arrive -> Wait (at Arrive time) -> ContinueWait (at server's next available serve time) -> _ContinueWait (at server's next available time, if customer is still in the queue)_ -> Serve (at server's next available serve time) -> Done (server time + service time) -> Rest (at done time).

I'm not sure what is the issue. Would appreciate any help!

## Screenshots
![image](https://user-images.githubusercontent.com/113164119/204473165-cdcf6c44-b4a2-4e45-82f9-888525c6a175.png)

![image](https://user-images.githubusercontent.com/113164119/204473208-237df3f8-f102-4481-87f0-9acfc59fbaf6.png)

**Comment by cyiting** [comment]
It could be something to do with the way you check whether or not the customer should continue waiting or be served next? The way I did it was to check whether my Wait event time corresponds to the server's free time.

**Comment by yqlok17** [comment]
I've tried to check but I don't see anything wrong with my code - my Wait will always return a ContinueWait at the corresponding server's next free time, and my ContinueWait will return a serve if customer can be served and dequeues the customer, and will return a new ContinueWait at the server's next serve time if customer cannot be served.

**Comment by cyiting** [comment]
I think i meant the mechanism to check whether they should wait or be served? Or maybe your server's next free time isn't updated in the Wait event?

**Comment by yqlok17** [comment]
Yep I think I've narrowed it down to a queue issue as the server free time looks okay.

**Comment by Nicklelodeon** [comment]
I'm not sure how u implemented the queue but the only change I made from previous labs to the queue was to create a queue manager that keeps track of all the queues (all self check will have the same queue). The implementation (when to add/remove from the queue) should be the same as previous labs
